app/controllers/mixins/actions/vm_actions/transform.rb

```diff
@@ -17,7 +17,8 @@ def vm_transform
             @right_cell_text = _("Transform VMs to RHV")
             simple_dialog_initialize(
               dialog.resource_actions.first,
-              :header => @right_cell_text
+              :header => @right_cell_text,
+              :target => ExtManagementSystem.find_by(:type => 'ManageIQ::Providers::Redhat::InfraManager')
```

Comment:
@smelamud 
If you use the different URL's you should be able to ExtManagementSystem.find_by(:id => ....)

Since you get passed the id just like when you do a single VM.

Reply:
@mkanoor What about Compute > Infrastructure > Virtual Machines? If the mass transform button is clicked there, what it receives? There is neither provider nor VM selected.